File: 1 - Algorithmic Toolbox/Week_3/1 - Money Change/money_change.py
import sys

def money_change( money: int ) -> int:

    coins = 0

    while( money != 0 ):
        if ( money >= 10 ):
            money -= 10
            coins += 1
        elif ( money >= 5 ):
            money -= 5
            coins += 1
        else:
            money -= 1
            coins += 1
    
    return coins

def main( ) -> None:
    money = int( input( ) )
    print( money_change( money ) )

if __name__ == "__main__":
    main( )

File: 1 - Algorithmic Toolbox/3 - Greedy Algorithms/1 - Money Change/money_change.c
#include <stdio.h>
#include <stdlib.h>

int money_change(int money)
{
    int coins = 0 ;

    while (money != 0) {
        if (money >= 10) {
            money -= 10 ;
        } else if (money >= 5) {
            money -= 5 ;
        } else {
            money -= 1 ;
        }
        coins++ ;
    }

    return coins ;
}

int main(void)
{
    int money ;
    scanf("%d", &money) ;

    printf("%d\n", money_change(money)) ;

    return 0 ;
}